### ISSUE TITLE
Classification

### ISSUE THREAD

**Issue by sctrueew** [ultralytics/yolov5#9090]
- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

Hi,

Thanks for great work, Is it possible to train a model with efficientnet_v2?

**Comment by AyushExel** [comment]
@sctrueew yes you can train all yolov5-cls models from n to x and all models available via torchvision.
I see that these are the efficientnet_v2 models available on torchvision https://pytorch.org/vision/stable/models/efficientnetv2.html . Let us know if you face any problems

**Comment by sctrueew** [comment]
I use this model for training "efficientnet_v2_s" but the training loss is nan and top1_acc is 0.6 in all epochs.

model: efficientnet_v2_s
batch: 16 and 32 tested
img: 256


   Epoch   GPU_mem  train_loss   test_loss    top1_acc    top5_acc
      1/10     1.65G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.11it/s]
      2/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:13<00:00,  1.14it/s]
      3/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.13it/s]
      4/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.14it/s]
      5/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:13<00:00,  1.15it/s]
      6/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.14it/s]
      7/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.14it/s]
      8/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.14it/s]
      9/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.13it/s]
     10/10     1.74G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:14<00:00,  1.14it/s]

**Comment by glenn-jocher** [comment]
@sctrueew efficientnet_v2_s  works correctly:

<img width="898" alt="Screenshot 2022-08-22 at 22 23 22" src="https://user-images.githubusercontent.com/26833433/186012120-fe317885-c7d7-4fd5-a88c-f79557190bf5.png">

We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
- ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by sctrueew** [comment]
@glenn-jocher @AyushExel Yes, that is correct. It worked when I downgraded PyTorch from 1.12.1 to 1.10.2.

**Comment by sctrueew** [comment]
@AyushExel Hi, There is a problem with the latest version of PyTorch('1.12.1'). I got this result for all models.

    2/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.39it/s]
    3/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.35it/s]
    4/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.56it/s]
    5/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.57it/s]
    6/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.55it/s]
    7/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.50it/s]
    8/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.50it/s]
    9/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.49it/s]
    10/10     0.43G         nan         nan         0.6           1: 100%|██████████| 16/16 [00:02<00:00,  5.57it/s]

**Comment by AyushExel** [comment]
@sctrueew so its working on torch 1.10.2 ?

**Comment by sctrueew** [comment]
@AyushExel yes, I created a new env and installed all requirements in the repo but I still get nan

OS: win10
PyTorch: 1.12.1

I moved to the Colab with PyTorch (1.12.1) It's no problem. Do you have any idea?

**Comment by AyushExel** [comment]
@sctrueew If its not reproducible on a colab then it'll be hard to debug. It might end up being a windows-pytorch specific problem.

**Comment by glenn-jocher** [comment]
@sctrueew YOLOv5 CI runs every 24 hours on all operating systems for all tasks. Classification training on Windows with latest torch stable 1.12.1 works correctly:
https://github.com/ultralytics/yolov5/runs/7963711856?check_suite_focus=true

**Comment by sctrueew** [comment]
@glenn-jocher Yes that's correct but in CI torch is using CPU not GPU. I moved to latest version of the torch in CPU mode it worked but the torch(1.12.1) GPU not working. I downgraded the torch version to 1.10 GPU works fine.

Collecting torch>=1.7.0
  Using cached https://download.pytorch.org/whl/cpu/torch-1.12.1%2Bcpu-cp310-cp310-win_amd64.whl (161.7 MB)
Collecting torchvision>=0.8.1
  Using cached https://download.pytorch.org/whl/cpu/torchvision-0.13.1%2Bcpu-cp310-cp310-win_amd64.whl (1.1 MB)

**Comment by sctrueew** [comment]
I debugged with the torch (1.12.1+cud11.6) and get nan in line 
https://github.com/ultralytics/yolov5/blob/d0fa0042bd7775b2dd191d66548f5d8b677bb756/classify/train.py#L199

**Comment by sctrueew** [comment]
this is my configs:

> C:\Users>nvcc --version
> nvcc: NVIDIA (R) Cuda compiler driver
> Copyright (c) 2005-2021 NVIDIA Corporation
> Built on Fri_Dec_17_18:28:54_Pacific_Standard_Time_2021
> Cuda compilation tools, release 11.6, V11.6.55
> Build cuda_11.6.r11.6/compiler.30794723_0

--------------------------------------------------------------

> import torch
torch.__version__
  '1.12.1+cu116'
torch.version.cuda
  '11.6'
torch.cuda.is_available()
  True
torch.cuda.device_count()
  1
torch.cuda.get_device_name(0)
  'NVIDIA GeForce GTX 1650 with Max-Q Design'

**Comment by glenn-jocher** [comment]
> @sctrueew efficientnet_v2_s works correctly:
> 
> <img alt="Screenshot 2022-08-22 at 22 23 22" width="898" src="https://user-images.githubusercontent.com/26833433/186012120-fe317885-c7d7-4fd5-a88c-f79557190bf5.png">
> 
> We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.
> 
> ## How to create a Minimal, Reproducible Example
> When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:
> 
> * ✅ **Minimal** – Use as little code as possible to produce the problem
> * ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
> * ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem
> 
> For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:
> 
> * ✅ **Current** – Verify that your code is up-to-date with GitHub [master](https://github.com/ultralytics/yolov5/tree/master), and if necessary `git pull` or `git clone` a new copy to ensure your problem has not already been solved in [master](https://github.com/ultralytics/yolov5/tree/master).
> * ✅ **Unmodified** – Your problem must be reproducible using official YOLOv5 code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.
> 
> If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/yolov5/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem.
> 
> Thank you! 😃

We don't have windows systems to reproduce, but since this is local to your system your environment is likely the cause.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Google Colab and Kaggle** notebooks with free GPU: <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
